ci/playbooks/kuttl/deploy-deps.yaml

```diff
@@ -0,0 +1,136 @@
+      when:
+        - cifmw_operator_build_output is defined
+        - "'operators' in cifmw_operator_build_output"
+
+    - name: Set install_yamls Makefile environment variables
+      vars:
+        operators_build_output: "{{ (cifmw_operator_build_output | default({'operators':{}})).operators }}"
+        _openstack_operator_images:
+          OPENSTACK_IMG: "{{ cifmw_operator_build_output.operators['openstack-operator']['image_catalog'] | default(omit) }}"
+          OPENSTACK_BUNDLE_IMG: "{{ cifmw_operator_build_output.operators['openstack-operator']['image_bundle'] | default(omit) }}"
+      ansible.builtin.set_fact:
+        cifmw_edpm_prepare_common_env: >-
+          {{
+            cifmw_install_yamls_environment |
+            combine({'PATH': cifmw_path}) |
+            combine(cifmw_edpm_prepare_extra_vars | default({}))
+          }}
+        cifmw_edpm_prepare_operators_build_output: "{{ operators_build_output }}"
+        cifmw_edpm_prepare_make_openstack_env: "{{ _local_operators_indexes | combine(_openstack_operator_images) }}"
+
+    - name: Deploy openstack Operators
+      environment:
+        KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+        PATH: "{{ cifmw_path | default(ansible_env.PATH) }}"
+      block:
+        - name: detect if openstack operator is installed
+          ansible.builtin.command:
+            cmd: "oc get sub --ignore-not-found=true -n openstack-operators -o name openstack-operator"
+          ignore_errors: true
+          register: openstack_operator_subscription
+
+        - name: Install openstack operator
+          when: openstack_operator_subscription.stdout == ""
+          vars:
+            make_openstack_env: "{{ cifmw_edpm_prepare_common_env |
+              combine(cifmw_edpm_prepare_make_openstack_env) }}"
+            make_openstack_dryrun: "{{ cifmw_edpm_prepare_dry_run }}"
+          ansible.builtin.include_role:
+            name: 'install_yamls_makes'
+            tasks_from: 'make_openstack'
+
+        - name: Wait for OpenStack subscription creation
+          when: openstack_operator_subscription.stdout == ""
+          ansible.builtin.command:
+            cmd: >-
+              oc get sub openstack-operator
+              --namespace=openstack-operators
+              -o=jsonpath='{.status.installplan.name}'
+          register: cifmw_edpm_prepare_wait_installplan_out
+          until: cifmw_edpm_prepare_wait_installplan_out.rc == 0 and cifmw_edpm_prepare_wait_installplan_out.stdout != ""
+          retries: 30
+          delay: 10
+
+        - name: Wait for OpenStack operator to get installed
+          when: openstack_operator_subscription.stdout == ""
+          ansible.builtin.command:
+            cmd: >-
+              oc wait InstallPlan {{ cifmw_edpm_prepare_wait_installplan_out.stdout }}
+              --namespace=openstack-operators
+              --for=jsonpath='{.status.phase}'=Complete --timeout=20m
+
```

Comment:
we may want to have a hook here to install operators that are not part of the meta operator or other customisation.

i.e. we need to be able to install watcher operator here after the OpenStack operator is installed.

the "make kuttl-test-prep" should not actually depend on the operator its testing being installed so this will still work as we can add a pre-run playbook to the job that inherits form this so its not a blocker.

we have a hookpoint as a result of that but just wanted to highlight this.

Reply:
just to illustrate how we do it in watcher-operator here is the hook to install it https://github.com/openstack-k8s-operators/watcher-operator/pull/14/files#diff-108978819c05ae183d88ec87959c2341a94cfc3f9465e3aeee82d554217b4f58R56

Reply:
ya that pretty clean so we likely don't need a hook point here.
we could always add one later if we hit issues with the current approch.

Reply:
the one thing we need to be careful of is the watcher operator will edepnd on CRD defined in the main operator bundels. i.e. the Mariadb CRDs so I'm not sure if the post_infa hook is after the oepsntack operators are installed.

with that said the side effect will be that they watcher-oeprator will fail until the other CRDs are defined but that is likely ok. it should heal when the rest of the operators are installed.